File: Android/FireClassification/app/src/main/java/com/thor/fireclassification/service/ApiClient.java
package com.thor.fireclassification.service;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApiClient {
    public static Retrofit retrofit;
    private static final String BASE_URL = "https://jovial-light-312714.et.r.appspot.com";

    public static Retrofit getRetrofitInstance(){
        if (retrofit == null){
            retrofit =  new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }

        return  retrofit;
    }
}

File: Android/FireApp/app/src/main/java/com/thor/firedetectionapp/service/ApiClient.kt
package com.thor.firedetectionapp.service

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class ApiClient {
    var retrofit: Retrofit? = null
    private val BASE_URL = "https://jovial-light-312714.et.r.appspot.com"

    fun getRetrofitInstance(): Retrofit? {
        if (retrofit == null) {
            retrofit = Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()
        }
        return retrofit
    }
}